src/cli.ts

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env node
+
+import { JSONSchema } from './JSONSchema'
+
+import { readFile, writeFile } from 'mz/fs'
+
+import { compile } from './index'
+
+// require it instead of import till typings for stdin are not released
+const stdin: (cb: (str: string) => void) => void = require('stdin')
+
+// require it instead of import till typings for minimist are kind of broken
+import minimist = require('minimist')
+
+const argv: minimist.ParsedArgs = minimist(process.argv.slice(2), {
+  alias: {
+    input: [ 'i' ],
+    output: [ 'o' ]
+  }
+})
+const argIn: string = argv._[0] || argv['input']
+const argOut: string = argv._[1] || argv['output']
+
+function readInput(): Promise<string> {
```

Comment:
I'm not sure it does make sense. The code could be simplified to
```
function readInput(): Promise<string> {
  if (!argIn) {
    return new Promise(stdin);
  }

  return readFile(argIn)
}
```
Is it the case for `async`?